src/sentry/event_manager.py

```diff
@@ -948,23 +948,25 @@ def _get_event_user_impl(project, data, metrics_tags):
     euser_id = cache.get(cache_key)
     if euser_id is None:
         metrics_tags["cache_hit"] = "false"
-        try:
-            with transaction.atomic(using=router.db_for_write(EventUser)):
-                euser.save()
-            metrics_tags["created"] = "true"
-        except IntegrityError:
-            metrics_tags["created"] = "false"
-            try:
-                euser = EventUser.objects.get(project_id=project.id, hash=euser.hash)
-            except EventUser.DoesNotExist:
-                metrics_tags["created"] = "lol"
-                # why???
-                e_userid = -1
-            else:
-                if euser.name != (user_data.get("name") or euser.name):
-                    euser.update(name=user_data["name"])
-                e_userid = euser.id
-            cache.set(cache_key, e_userid, 3600)
```

Comment:
Seems like there was a bug and we cached only EventUsers that were not created? The `test_event_user` test depended on that buggy behavior.